sdk/lib/atl/atlimage.h

```diff
@@ -766,13 +766,8 @@ class CImage
                 CString ext(pCodecs[i].FilenameExtension);
```

Comment:
(Random observation): It should be possible to collect both this for-loop for building the all-files extension list, together with the next for-loop, by:
1. always adding the NULL Guid into aguidFileTypes first in case pszAllFilesDescription != NULL (as it is currently done);
2. in the for-loop below, while building the strFilter string, also maintain an "all-extensions" string (the equivalent of the "extensions" string used in that pszAllFilesDescription != NULL case above);
3. Once the strFilter string is built from the for-loop, you prepend the pszAllFilesDescription plus the "all-extensions" string to strFilter.

The resulting strFilter should be correct.